src/main/java/org/jboss/ejb/client/ConfigurationBasedEJBClientContextSelector.java

```diff
@@ -168,6 +193,30 @@ private static void parseInterceptorsType(final ConfigurationXMLStreamReader str
     }
 
     private static void parseInterceptorType(final ConfigurationXMLStreamReader streamReader, final EJBClientContext.Builder builder) throws ConfigXMLParseException {
+        final Class<? extends EJBClientInterceptor> interceptorClass = parseClassType(streamReader, builder, EJBClientInterceptor.class);
+        builder.addInterceptor(interceptorClass);
+    }
+
+    private static void parseClusterNodeSelectorType(final ConfigurationXMLStreamReader streamReader, final EJBClientContext.Builder builder) throws ConfigXMLParseException {
+        final Class<? extends ClusterNodeSelector> clusterNodeSelectorClass = parseClassType(streamReader, builder, ClusterNodeSelector.class);
+        try {
+            builder.setClusterNodeSelector(clusterNodeSelectorClass.newInstance());
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new ConfigXMLParseException(e);
+        }
+    }
+
+    private static void parseDeploymentNodeSelectorType(final ConfigurationXMLStreamReader streamReader, final EJBClientContext.Builder builder) throws ConfigXMLParseException {
+        final Class<? extends DeploymentNodeSelector> deploymentNodeSelectorClass = parseClassType(streamReader, builder, DeploymentNodeSelector.class);
+        try {
+            builder.setDeploymentNodeSelector(deploymentNodeSelectorClass.newInstance());
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new ConfigXMLParseException(e);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> Class<? extends T> parseClassType(final ConfigurationXMLStreamReader streamReader, final EJBClientContext.Builder builder, Class<T> interfaceToImplement) throws ConfigXMLParseException {
```

Comment:
Is the `builder` parameter unused?

Reply:
Yes, it is, I'll remove it, just didn't notice, sorry...